lib/gradient/elixir_checker.ex

```diff
@@ -132,4 +155,30 @@ defmodule Gradient.ElixirChecker do
   end
 
   def remove_injected_forms(forms, false), do: forms
+
+  defp warn_missing_spec(to_filter, forms) do
+    all_fnas_specs =
+      Enum.reduce(forms, %{:fun => [], :spec => []}, fn
+        {:fun, {name, _}, _}, %{fun: fnas} = acc -> %{acc | fun: [name | fnas]}
+        {:spec, {name, _}, _}, %{spec: fnas} = acc -> %{acc | spec: [name | fnas]}
+        _, acc -> acc
+      end)
+
+    ret = (all_fnas_specs[:fun] -- all_fnas_specs[:spec]) -- to_filter
```

Comment:
```suggestion
    ret = (all_fnas_specs[:fun] -- all_fnas_specs[:spec]) -- to_filter
    ret = MapSet.new(ret)
```

Reply:
If we're repeatedly doing a lookup in this structure, it's worth making sure the lookup operation is efficient. Probably not a big deal in case of a few elements, but might make a huge impact in general.